ic-transport-types/src/lib.rs

```diff
@@ -117,6 +118,36 @@ pub struct ReadStateResponse {
     pub certificate: Vec<u8>,
 }
 
+/// The response from a request to the `call` endpoint.
+// TODO: Inline this enum. No need for it to be public.
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "status", rename_all = "snake_case")]
+pub enum TransportCallResponse {
+    /// A certified response.
+    Replied {
+        /// The CBOR serialized certificate for the call response.
+        #[serde_as(as = "Bytes")]
+        certificate: Vec<u8>,
+    },
+
+    /// A non replicated rejection from the replica.
+    NonReplicatedRejection(RejectResponse),
+
+    /// The replica timed out the sync request. The status of the request must be polled.
+    Accepted,
+}
+
+/// The response from a request to the `call` endpoint.
+#[derive(Debug, PartialEq, Eq)]
+pub enum CallResponse<Out> {
+    /// The call complted, and the response is available.
+    Response(Out),
+    /// The replica timed out the update call, and the request id should be used to poll for the response
+    /// using the `request_status` request type.
```

Comment:
Even though it is in ic-transport-types, this should reference an ic-agent function rather than a HTTP endpoint.